src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/3.sql

```diff
@@ -2105,67 +2097,66 @@ GO
 
 --
 -- STORED PROCEDURE
---     Updates an instance schema.
+--     Update an instance schema.
 --
 -- DESCRIPTION
 --     Modifies an existing record in the InstanceSchema table.
 --
 -- PARAMETERS
 --    @name
 --         * The unique name for a particular instance
---     @currentVersion
---         * The current version of the schema that the given instance is using
 --     @maxVersion
 --         * The maximum supported schema version for the given instance
 --     @minVersion
 --         * The minimum supported schema version for the given instance
 --
 CREATE PROCEDURE dbo.UpsertInstanceSchema
     @name varchar(64),
-    @currentVersion int,
     @maxVersion int,
     @minVersion int
     
 AS
     SET NOCOUNT ON
 
     DECLARE @timeout datetime2(0) = DATEADD(minute, 2, SYSUTCDATETIME())
-    IF EXISTS(SELECT *
+    DECLARE @currentVersion int = (SELECT COALESCE(MAX(Version), 0)
```

Comment:
If upsert is called when `MAX(SchemaVersion.Version)` is null, should we be throwing an error instead of setting the value to 0?

Reply:
I think this is just to take into account the first time when no schema version execution is complete. And we should not be throwing exception rather just upgrade to the available versions by the default upgrade or tool.

Reply:
Right, makes sense. Thanks 👍